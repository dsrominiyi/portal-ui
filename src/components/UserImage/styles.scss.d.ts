export const userPhoto: string;
export const defaultUserIcon: string;
