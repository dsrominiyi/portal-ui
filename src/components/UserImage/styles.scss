@import '../../style/vars';

.user-photo {
  height: 50px;
  border-radius: 50%;
}

.default-user-icon {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 50px;
  width: 50px;
  color: $brand-white;
  background-color: $brand-info;
  font-size: 1.4em;
  border-radius: 50%;
}
