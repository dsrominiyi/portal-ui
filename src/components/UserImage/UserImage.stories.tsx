import React from 'react';
import { storiesOf } from '@storybook/react';

import UserImage from '.';

import { smallerImg } from '../../../.storybook/styles.scss';

const stories = storiesOf('UserImage', module);

const render = (name: UserName, imageUrl?: string, additionalClasses?: string[]) => () => (
  <UserImage name={name} imageUrl={imageUrl} additionalClasses={additionalClasses} />
);

const name = { forename: 'John', surname: 'Smith' };
const imageUrl = '/images/person-who-does-not-exist.jpg';

export const storyRenders = {
  noPhoto: render(name),
  withPhoto: render(name, imageUrl),
  extraClass: render(name, imageUrl, [smallerImg])
};

stories.add('With no profile photo', storyRenders.noPhoto);
stories.add('With a profile photo', storyRenders.withPhoto);
stories.add('With additional styling', storyRenders.extraClass);
