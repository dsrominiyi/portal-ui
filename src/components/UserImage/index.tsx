import React, { FunctionComponent } from 'react';

import styles from './styles.scss';

interface Props {
  name: UserName;
  imageUrl?: string;
  additionalClasses?: string[];
}

export const UserImage: FunctionComponent<Props> = ({
  name,
  imageUrl,
  additionalClasses = []
}: Props) => {
  const fullName = `${name.forename} ${name.surname}`;

  const imgClasses = [styles.userPhoto, ...additionalClasses];
  const iconClasses = [styles.defaultUserIcon, ...additionalClasses];

  const userImage = (
    <img src={imageUrl} alt={fullName} title={fullName} className={imgClasses.join(' ')} />
  );
  const defaultIcon = (
    <span className={iconClasses.join(' ')}>
      {`${name.forename.charAt(0)}${name.surname.charAt(0)}`}
    </span>
  );

  return imageUrl ? userImage : defaultIcon;
};

UserImage.displayName = 'UserImage';
export default UserImage;
