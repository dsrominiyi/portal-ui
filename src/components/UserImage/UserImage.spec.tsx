import { storyRenders } from './UserImage.stories';

import testStorySnapshots from '../../helpers/test/testStorySnapshots';

describe('UserImage', () => {
  test('component story renders match snapshots', () => {
    testStorySnapshots(storyRenders);
  });
});
