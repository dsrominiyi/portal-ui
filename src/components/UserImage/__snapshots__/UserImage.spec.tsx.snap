// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserImage component story renders match snapshots: extraClass 1`] = `
<img
  alt="John Smith"
  className="userPhoto smallerImg"
  src="/images/person-who-does-not-exist.jpg"
  title="John Smith"
/>
`;

exports[`UserImage component story renders match snapshots: noPhoto 1`] = `
<span
  className="defaultUserIcon"
>
  JS
</span>
`;

exports[`UserImage component story renders match snapshots: withPhoto 1`] = `
<img
  alt="John Smith"
  className="userPhoto"
  src="/images/person-who-does-not-exist.jpg"
  title="John Smith"
/>
`;
